libraries/botbuilder-ai/tests/luisRecognizer.test.js

```diff
@@ -87,57 +92,59 @@ class OverrideFillRecognizer extends LuisRecognizer {
     }
 }
 
-function throttle(callback) {
-    if (mockLuis) {
-        callback();
-    } else {
-        // If actually calling LUIS, need to throttle our requests
-        setTimeout(callback, 1000);
-    }
+async function throttle() {
+    return new Promise((resolve) => {
+        return mockLuis ? resolve() : setTimeout(resolve, 1000);
+    });
 }
 
-function WithinDelta(token1, token2, delta, compare) {
-    var within = true;
-    if (token1 == null || token2 == null) {
-        within = token1 == token2;
-    } else if (Array.isArray(token1) && Array.isArray(token2)) {
-        within = token1.length == token2.length;
-        for (var i = 0; within && i < token1.length; ++i) {
-            within = WithinDelta(token1[i], token2[i], delta, compare);
+function isWithinDelta(left, right, delta, compare) {
+    let within = true;
+
+    if (left == null || right == null) {
+        within = left === right;
+    } else if (Array.isArray(left) && Array.isArray(right)) {
+        within = left.length === right.length;
+        for (let i = 0; within && i < left.length; i++) {
+            // eslint-disable-next-line security/detect-object-injection
+            within = isWithinDelta(left[i], right[i], delta, compare);
         }
-    } else if (typeof token1 === 'object' && typeof token2 === 'object') {
-        Object.keys(token2).forEach((k) => token2[k] === undefined && delete token2[k]);
```

Comment:
This _could_ have caused failures where `left` had undefined values that might have had matching undefined values in `right`. I'm pretty sure this fixed the unhandled promise rejection error that's been getting logged during recent test runs.